src/Bicep.LangServer.IntegrationTests/BuildCommandTests.cs

```diff
@@ -91,5 +91,99 @@ public async Task Build_command_should_generate_template_with_symbolic_names_if_
+                },
+                {
+                    "moduleB.bicep",
+                    """
+                    import * as typesA from 'moduleA.bicep'
+                    @export()
+                    type typeB = {
+                        optionsA: typesA.typeA
+                        propB: string
+                    }
+                    """
+                },
+                {
+                    "moduleC.bicep",
+                    """
+                    import * as typesA from 'moduleA.bicep'
+                    @export()
+                    type typeC = {
+                        optionsA: typesA.typeA
+                        propC: string
+                    }
+                    """
+                },
+            };
+
+            var outputDirectory = FileHelper.GetUniqueTestOutputPath(TestContext);
+            Directory.CreateDirectory(outputDirectory);
+
+            foreach (var kvp in files)
+            {
+                using var file = File.CreateText(Path.Combine(outputDirectory, kvp.Key));
+                await file.WriteAsync(kvp.Value);
+            }
+
+            var diagnosticsListener = new MultipleMessageListener<PublishDiagnosticsParams>();
+
+            using var helper = await LanguageServerHelper.StartServer(
+                this.TestContext,
+                options => options.OnPublishDiagnostics(diagnosticsListener.AddMessage),
+                services => services
+                    .WithNamespaceProvider(BuiltInTestTypes.Create())
+                    .WithFeatureOverrides(new(TestContext)));
+            var client = helper.Client;
+
+            var mainPath = Path.Combine(outputDirectory, "main.bicep");
+            var moduleCPath = Path.Combine(outputDirectory, "moduleC.bicep");
+
+            // First, load main.bicep (and compile the transitive closure)
+            client.TextDocument.DidOpenTextDocument(
+                TextDocumentParamHelper.CreateDidOpenDocumentParamsFromFile(mainPath, 1));
+            await diagnosticsListener.WaitNext();
+
+            // Second, load moduleC.bicep (and compile its transitive closure)
+            // Because main.bicep and moduleC.bicep exist in separate "active contexts" in the language server, this
+            // sequence of actions will result in the compilation manager having two separate SemanticModels for
+            // moduleA.bicep: one in the transitive closure of main.bicep (still reachable via the import of
+            // moduleB.bicep) and another in the transitive closure of moduleC.bicep.
+            client.TextDocument.DidOpenTextDocument(
+                TextDocumentParamHelper.CreateDidOpenDocumentParamsFromFile(moduleCPath, 1));
+            await diagnosticsListener.WaitNext();
+
+            // Make sure we can still build main.bicep event though the `typeA` symbol from moduleA.bicep has two
+            // instantiations (and two different memory addresses) within the compilation manager.
```

Comment:
Do you know why this is happening? Within a single compilation, I would always expect each file to have a single semantic model representing it.